lib/upgrade/migrator/migrate.go

```diff
@@ -655,6 +657,74 @@ func (m *migrationHelper) queryAndConvertV1ToV3Nodes(data *MigrationData) error
 	return nil
 }
 
+// ShouldMigrate checks version information and reports if migration is needed
+// and is possible.
+func (m *migrationHelper) ShouldMigrate() (bool, error) {
+	ci, err := m.clientv3.ClusterInformation().Get(context.Background(), "default", options.GetOptions{})
+	if err == nil {
+		// Check the version to see if maybe the migration has been started
+		// but not finished.
+		if canMigrate(ci.Spec.CalicoVersion) {
```

Comment:
nit:  unless you're planning to add logs for each branch, this can be contracted to:

`return canMigrate(), nil`